### ISSUE TITLE
(wmapro) Failed audio resync.

### ISSUE THREAD

**Issue by tung9484** [mpv-player/mpv#2735]
PS C:\Users\VTT> D:\Software\MPV\mpv --no-config "D:\Videos\X-Files\Risa Tachibana\STAR-467A.wmv"
Playing: D:\Videos\X-Files\Risa Tachibana\STAR-467A.wmv
 (+) Video --vid=1 (vc1)
 (+) Audio --aid=1 --alang=jpn (wmapro)
File tags:
 Artist: DMM
AO: [wasapi] 48000Hz stereo 2ch float
VO: [opengl] 1920x1080 yuv420p
Failed audio resync.
(Paused) AV: 00:00:03 / 02:08:34 (0%) A-V:  0.000 ct: -0.344

mediainfo
General
Complete name                  : D:\Videos\X-Files\Risa Tachibana\STAR-467A.wmv
Format                         : Windows Media
File size                      : 5.38 GiB
Duration                       : 2h 8mn
Overall bit rate mode          : Constant
Overall bit rate               : 5 993 Kbps
Maximum Overall bit rate       : 6 048 Kbps
Performer                      : DMM
Encoded date                   : UTC 2013-08-23 08:07:30.345
Copyright                      : SODクリエイト

Video
ID                             : 2
Format                         : VC-1
Format profile                 : Advanced@L3
Codec ID                       : WVC1
Codec ID/Hint                  : Microsoft
Description of the codec       : Windows Media Video 9 Advanced Profile
Duration                       : 2h 8mn
Bit rate mode                  : Constant
Bit rate                       : 5 744 Kbps
Width                          : 1 920 pixels
Height                         : 1 080 pixels
Display aspect ratio           : 16:9
Frame rate                     : 29.970 (29970/1000) fps
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Scan type                      : Progressive
Compression mode               : Lossy
Bits/(Pixel*Frame)             : 0.092
Stream size                    : 5.16 GiB (96%)
Language                       : Japanese

Audio
ID                             : 1
Format                         : WMA
Format profile                 : Pro
Codec ID                       : 162
Codec ID/Info                  : Windows Media Audio
Description of the codec       : Windows Media Audio 10 Professional - 256 kbps, 44 kHz, 2 channel 16 bit (A/V) 1-pass CBR
Duration                       : 2h 8mn
Bit rate mode                  : Constant
Bit rate                       : 256 Kbps
Channel(s)                     : 2 channels
Sampling rate                  : 44.1 KHz
Bit depth                      : 16 bits
Stream size                    : 235 MiB (4%)
Language                       : Japanese

**Comment by kevmitch** [comment]
Can you provide a smallish sample file?

**Comment by laichiaheng** [comment]
I have the same problem with the same kind of video, too.

**Comment by kevmitch** [comment]
above request for sample still stands

**Comment by tp0** [comment]
Same here. Is the problem VMA Pro audio codec?

I could provide a sample in theory but as in OP's case, the problem files I've encountered are also all copyrighted adult videos, so I'm not sure if I should do this.

This does not happen in mpv-fork Bomi (0.9.11).

mpv 17.0:
```
mpv git-449b948 (C) 2000-2016 mpv/MPlayer/mplayer2 projects
 built on Thu May 19 11:12:22 UTC 2016
ffmpeg library versions:
   libavutil       55.24.100
   libavcodec      57.40.100
   libavformat     57.36.100
   libswscale      4.1.100
   libavfilter     6.45.100
   libswresample   2.0.101
ffmpeg version: N-79883-g38eeb85
```

Mediainfo:
```
Video
ID                                       : 1
Format                                   : VC-1
Format profile                           : Advanced@L3
Codec ID                                 : WVC1
Codec ID/Hint                            : Microsoft
Description of the codec                 : vc1
Duration                                 : 3mn 0s
Bit rate                                 : 5 534 Kbps
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate                               : 30.000 fps
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Compression mode                         : Lossy
Bits/(Pixel*Frame)                       : 0.089
Stream size                              : 119 MiB (92%)

Audio
ID                                       : 2
Format                                   : WMA
Format profile                           : Pro
Codec ID                                 : 162
Codec ID/Info                            : Windows Media Audio
Description of the codec                 : wmapro
Duration                                 : 3mn 0s
Bit rate                                 : 256 Kbps
Channel(s)                               : 2 channels
Sampling rate                            : 44.1 KHz
Bit depth                                : 16 bits
Stream size                              : 5.49 MiB (4%)
```

Another video:
```
Video
ID                                       : 2
Format                                   : VC-1
Format profile                           : MP@HL
Codec ID                                 : WMV3
Codec ID/Info                            : Windows Media Video 9
Codec ID/Hint                            : WMV3
Description of the codec                 : Windows Media Video 9 - Professional
Duration                                 : 7mn 38s
Bit rate mode                            : Constant
Bit rate                                 : 5 000 Kbps
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate                               : 25.000 fps
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Compression mode                         : Lossy
Bits/(Pixel*Frame)                       : 0.096
Stream size                              : 274 MiB (93%)
Language                                 : English (US)
NumberOfFrames                           : 11462

Audio
ID                                       : 1
Format                                   : WMA
Format profile                           : Pro
Codec ID                                 : 162
Codec ID/Info                            : Windows Media Audio
Description of the codec                 : Windows Media Audio 10 Professional - 440 kbps, 44 kHz, 2 channel 24 bit 2-pass VBR
Duration                                 : 7mn 38s
Bit rate mode                            : Variable
Bit rate                                 : 410 Kbps
Channel(s)                               : 2 channels
Sampling rate                            : 44.1 KHz
Bit depth                                : 24 bits
Stream size                              : 22.4 MiB (8%)
Language                                 : English (US)
NumberOfFrames                           : 1146
```

**Comment by richardpl** [comment]
Could you try another asf_o demuxer?

**Comment by kevmitch** [comment]
> I could provide a sample but as in OP's case, the problem files I've encountered are also all copyrighted adult videos 

I think this qualifies as fair use, particularly if you can extract only a portion. If you are really paranoid, you can delete the link after a developer has confirmed they have it.

**Comment by tp0** [comment]
> Could you try another asf_o demuxer?

ELI5? Does that require building?

---

No need for sample after all, I found these old movie trailers:

http://streamingmovies.ign.com/movies/article/999/999907/worlds_grt_dad_063009_wmvhd.wmv
http://streamingmovies.ign.com/movies/article/109/1093150/jonahhex_trl2_052510_wmvhd.wmv

If it is not immediatley noticiable try exacerbating it by time seeking. This error shows up regardless:

```
$ mpv jonahhex_trl2_052510_wmvhd.wmv 
Playing: jonahhex_trl2_052510_wmvhd.wmv
 (+) Video --vid=1 (wmv3)
 (+) Audio --aid=1 --alang=eng (wmapro)
libva info: VA-API version 0.39.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
AO: [pulse] 48000Hz stereo 2ch float
Using hardware decoding (vaapi).
VO: [opengl-hq] 1280x544 vaapi
Failed audio resync.
AV: 00:00:06 / 00:01:51 (5%) A-V:  0.000 ct: -0.624

Audio/Video desynchronisation detected! Possible reasons include too slow
hardware, temporary CPU spikes, broken drivers, and broken files. Audio
position will not match to the video (see A-V status field).

AV: 00:00:17 / 00:01:51 (15%) A-V:  0.003 ct:  0.083
```

OP seems to use a different AO than I am using (pulse vs. wasapi) and OP is not using VA-API.

**Comment by tp0** [comment]
#3297 was same bug and apparently this was fixed accidentally (?). I no longer get the error and audio is synced with the above two trailer samples.

**Comment by wm4** [comment]
k
